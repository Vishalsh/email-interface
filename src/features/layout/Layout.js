import React from 'react';
import { Route, Switch } from 'react-router-dom';

import routes from 'constants/routes';
import Header from 'features/header/Header';
import Sidebar from 'features/sidebar/Sidebar';
import Mailbox from 'pages/mailbox/Mailbox';

import classes from './Layout.module.scss';

const Layout = () => (
  <>
  <div className={classes.header}>
    <Header />
  </div>
  <div className={classes.sidebar}>
    <Sidebar />
  </div>

  <main className={classes.main}>
    <Switch>
      <Route exact path="/" component={Mailbox}/>
      <Route path={routes.DASHBOARD} component={() => <h1>DASHBOARD</h1>}/>
      <Route path={routes.LAYOUTS} component={() => <h1>LAYOUTS</h1>}/>
      <Route path={routes.GRAPHS} component={() => <h1>GRAPHS</h1>}/>
      <Route path={routes.MAILBOX} component={Mailbox}/>
      <Route path={routes.METRICS} component={() => <h1>METRICS</h1>}/>
      <Route path={routes.WIDGETS} component={() => <h1>WIDGETS</h1>}/>
      <Route path={routes.FORMS} component={() => <h1>FORMS</h1>}/>
      <Route path={routes.APP_VIEWS} component={() => <h1>APP VIEWS</h1>}/>
    </Switch>
  </main>
  </>
);

export default Layout;
