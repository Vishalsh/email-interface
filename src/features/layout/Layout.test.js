import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Layout from './Layout';

describe('Layout', () => {
  it('should render the component', () => {
    expect(toJson(shallow(<Layout/>))).toMatchSnapshot();
  })
});