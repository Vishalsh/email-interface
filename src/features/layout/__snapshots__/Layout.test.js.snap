// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the component 1`] = `
<Fragment>
  <div
    className="header"
  >
    <Header />
  </div>
  <div
    className="sidebar"
  >
    <Sidebar />
  </div>
  <main
    className="main"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/dashboard"
      />
      <Route
        component={[Function]}
        path="/layouts"
      />
      <Route
        component={[Function]}
        path="/graphs"
      />
      <Route
        component={[Function]}
        path="/mailbox"
      />
      <Route
        component={[Function]}
        path="/metrics"
      />
      <Route
        component={[Function]}
        path="/widgets"
      />
      <Route
        component={[Function]}
        path="/forms"
      />
      <Route
        component={[Function]}
        path="/app-views"
      />
    </Switch>
  </main>
</Fragment>
`;
